File: web/src/layout/ControlsDemoXml.ts
//
// ControlsDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import htmlForDefault from './controls_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.ControlsDemoXml
export class ControlsDemoXml {
    xmlRoot!: HTMLElement;
    editableText!: HTMLInputElement;
    numberText!: HTMLInputElement;
    editableAutoText!: HTMLInputElement;
    editableTextCopy!: HTMLElement;
    editableTextBig!: HTMLTextAreaElement;
    spinner!: HTMLSelectElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.editableText = getViewById<HTMLInputElement>(view, "editableText");
        this.numberText = getViewById<HTMLInputElement>(view, "numberText");
        this.editableAutoText = getViewById<HTMLInputElement>(view, "editableAutoText");
        this.editableTextCopy = getViewById<HTMLElement>(view, "editableTextCopy");
        this.editableTextBig = getViewById<HTMLTextAreaElement>(view, "editableTextBig");
        this.spinner = getViewById<HTMLSelectElement>(view, "spinner");
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/ControlsDemoXml.kt
//
// ControlsDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class ControlsDemoXml {

    lateinit var editableText: EditText
    lateinit var numberText: EditText
    lateinit var editableAutoText: AutoCompleteTextView
    lateinit var editableTextCopy: TextView
    lateinit var editableTextBig: com.lightningkite.butterfly.views.widget.MultilineEditText
    lateinit var spinner: Spinner
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.controls_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        editableText = view.findViewById<EditText>(R.id.editableText)
        numberText = view.findViewById<EditText>(R.id.numberText)
        editableAutoText = view.findViewById<AutoCompleteTextView>(R.id.editableAutoText)
        editableTextCopy = view.findViewById<TextView>(R.id.editableTextCopy)
        editableTextBig = view.findViewById<com.lightningkite.butterfly.views.widget.MultilineEditText>(R.id.editableTextBig)
        spinner = view.findViewById<Spinner>(R.id.spinner)
        
        
        
        return view
    }
}